src/components/Calculator.js

```diff
@@ -1,34 +1,54 @@
 import React from 'react';
+import calculate from '../logic/calculate';
+import Button from './Button';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      calcObj: {
+        total: null,
+        next: null,
+        operation: null,
+      },
+    };
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(buttonName) {
+    const { calcObj } = this.state;
+    this.setState({
```

Comment:
- [ ] When I click one number I should see that number be displayed on the screen. Currently for example, if I click 8 the number does not appear on the screen, it appears only after I click + or -. The end result is being calculated correctly but I think you should work on the user experience. The user should be able to too see the numbers that he/she is clicking

Reply:
Thanks. I updated the code and fixed this issue.